src/averageAges.js

```diff
@@ -55,7 +67,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const hasMotherData = people.filter((person) => {
+    const motherInList = people.some((mother) => mother.name === person.mother);
+
+    return onlyWithSon ? person.sex === 'm' && motherInList : motherInList;
+  });
+
+  const avgAgeDiff = hasMotherData.reduce((sumAvg, person) => {
+    const motherBorn = people.find((mother) => mother.name === person.mother)
+      .born;
+
+    return avaregeAge(person.born, motherBorn, hasMotherData.length, sumAvg);
+  }, 0);
+
+  return avgAgeDiff;
+}
+
+function avaregeAge(minuendAge, subtrahendAge, persons, prev) {
+  return (minuendAge - subtrahendAge) / persons + prev;
 }
```

Comment:
This function calculates a number from the passed arguments. The name of this function must begin with a verb.
For example: `calculateAverageAge`, `getAverageAge`


You can calculate the age difference ahead of time and pass those values ​​as an array to this function
This function should only accept an array of numbers and a length for the divisor and calculate average.

Reply:
👌 fixed